src/test/regress/expected/multi_replicate_reference_table.out

```diff
@@ -464,7 +464,11 @@ SELECT create_reference_table('replicate_reference_table_insert');
 BEGIN;
 INSERT INTO replicate_reference_table_insert VALUES(1);
 SELECT 1 FROM master_add_node('localhost', :worker_2_port);
-ERROR:  cannot open new connections after the first modification command within a transaction
+ ?column?
```

Comment:
This is an interesting side effect. Let's think a little more if this is correct

Reply:
It seems like previously we had `SetUpDistributedTableDependencies `  while syncing objects: https://github.com/citusdata/citus/blob/885601c02caed52c14683b3eba5fed01111428d6/src/backend/distributed/metadata/node_metadata.c#L590

I'm in favor of keeping `EnsureNoModificationsHaveBeenDone` at `ActivateNode ()`, as otherwise we might have weird edge cases (may be not, but stil lets not enable it :))